src/pgduckdb_options.cpp

```diff
@@ -175,6 +180,39 @@ DuckdbCacheObject(Datum object, Datum type) {
 	return true;
 }
 
+typedef struct CacheInfo {
+	std::string cache_key;
+	std::string remote_path;
+	std::string file_size;
+} CacheInfo;
+
+static std::vector<CacheInfo>
+DuckdbGetCachedFile() {
+	std::string ext(".meta");
+	std::vector<CacheInfo> cache_info;
+	for (auto &p : std::filesystem::recursive_directory_iterator(CreateOrGetDirectoryPath("duckdb_cache"))) {
+		if (p.path().extension() == ext) {
+			std::ifstream cache_file_metadata(p.path());
+			std::string metadata;
+			std::vector<std::string> metadata_tokens;
+			while (std::getline(cache_file_metadata, metadata, ',')) {
+				metadata_tokens.push_back(metadata);
```

Comment:
Should we throw if `metadata_tokens.size() > 3` to avoid reading a potentially invalid file?

Reply:
We want to report back to user which metadata are invalid and print back ones that are good.